docs/plugins.md

```diff
@@ -84,27 +86,217 @@ const giphy = {
 - `.openConfigFile()`: Open the plugin config file in the user’s editor.
-- `.cancel()`: Indicate that the plugin operation canceled for some reason. [Example.](https://github.com/wulkano/kap/blob/efc32d12f381615c9fcfc41065d9c2ee200e8975/app/src/main/save-file-service.js#L28-L31) This closes the Kap export window. If the cancelation was not the result of a user gesture, use `.notify()` to inform the user why it was canceled.
+- `.cancel()`: Indicate that the plugin operation canceled for some reason. [Example.](https://github.com/wulkano/kap/blob/efc32d12f381615c9fcfc41065d9c2ee200e8975/app/src/main/save-file-service.js#L28-L31) If the cancelation was not the result of a user gesture, use `.notify()` to inform the user why it was canceled.
+- `.waitForDeepLink()`: Returns a Promise that resolves when a deep link for this plugin is opened. The link should be in the format `kap://plugins/{pluginName}/{rest}`, where `pluginName` is the npm package name and `rest` is the string the Promise will resolve with. This is useful for [OAuth flows](#oauth).
+
+#### Notes
+
+Use `context.setProgress()` whenever possible to keep the user updated on what's happening. The `.filePath()` method sets its own progress, so you should not do it for that step.
+
+Example plugins: [`kap-giphy`](https://github.com/wulkano/kap-giphy/blob/master/index.js), [`kap-s3`](https://github.com/SamVerschueren/kap-s3), [`kap-imgur`](https://github.com/kevva/kap-imgur), [`kap-streamable`](https://github.com/kevva/kap-streamable)
+
+### Edit services
+
+Only supported in Kap >= 3.2.0.
+
+An edit service lets you add an entry to the edit menu in the Kap editor window and process the recording before it gets converted and exported. The edit service receives an `mp4` file which is generated from the recording after trimming the duration and adjusting the size. It's expected to produce another `mp4` file at the given output location, which will then be passed to the appropriate share service.
+
+The edit service is a plain object defining some metadata:
+
+- `title`: The title used in the export menu. For example: `Reverse`.\
+  The text should be in [title case](https://capitalizemytitle.com), for example, `Slow Down`, not `Slow down`.
+- `configDescription`: A description displayed at the top of the configuration window. You can use this to explain the config options or link to API docs. Any links in this description will be parsed into clickable links automatically.
+- `action`: The function that is run when the user clicks the menu item. [Read more below.](#action)
+- `config`: Definition of the config the plugins needs. [Read more below](#config).
+
+The `config` and `configDescription` properties are optional.
+
+Example:
+
+```js
+const action = async context => {
+	// Do something
+
+	context.notify('Notify about something');
```

Comment:
This and some other examples are using tab-indentation. You should be consistent and use 2-space indentation for all examples.